parser.py

```diff
@@ -194,32 +194,58 @@ def download_file_if_larger(url, filename, index, count, sleep_time):
        Returns whether the file is now known to be the largest available, and the number of bytes downloaded.
     """
     requests = import_module('requests')
+    imagesize = import_module('imagesize')
+
+    pref = f'{index:3d}/{count:3d} {filename}: '
     # Sleep briefly, in an attempt to minimize the possibility of trigging some auto-cutoff mechanism
     if index > 1:
-        print(f'{index}/{count}: Sleeping...', end='\r')
+        print(f'{pref}Sleeping...', end='\r')
         time.sleep(sleep_time)
     # Request the URL (in stream mode so that we can conditionally abort depending on the headers)
-    print(f'{index}/{count}: Requesting headers for {url}...', end='\r')
-    size_before = os.path.getsize(filename)
+    print(f'{pref}Requesting headers for {url}...', end='\r')
+    byte_size_before = os.path.getsize(filename)
     try:
         with requests.get(url, stream=True) as res:
             if not res.status_code == 200:
                 raise Exception('Download failed')
```

Comment:
Now that we're printing the exceptions, please add the res.status code (and whatever else might be useful) here.

Reply:
I've done this now. I added status code, status text and text content of the response. Twitters responses for 404 are always empty, but I think it's nice to keep this code, so that content is displayed if Twitter will ever start to return error messages.